### ISSUE TITLE
Kernel Arguments Exceeding Limit

### ISSUE THREAD

**Issue by wolfzhu** [plaidml/plaidml#1630]
Hello,

I was trying to setup a toy module here including a LSTM layer. Simply here it is:

>>> from keras import layers
>>> from keras import Model
>>> inp = layers.Input(shape = (120, 1, ))
>>> x = layers.LSTM(120)(x)
>>> x = layers.LSTM(120)(inp)
>>> x = layers.Dense(32)(x)
>>> x = layers.Dense(1)(x)
>>> oup = x
>>> mod = Model(inputs = [inp], outputs = [oup])
>>> mod.compile(optimizer='adam', loss='mse')

Train set:
>>> import numpy as np
>>> inp = np.random.rand(1000, 120, 1)
>>> oup = np.random.rand(1000)

Training command:
>>> mod.fit(inp, oup, verbose = True, batch_size = 8, epochs = 10, validation_split = 0.3)

And I got the error like this:
Train on 700 samples, validate on 300 samples
Epoch 1/10
WARNING:plaidml:Failed to build program kernel_1715
WARNING:plaidml:Failed build log:
error: Total size of kernel arguments exceeds limit! Total arguments size: 3080, limit: 1024
error: backend compiler failed build.

-- Seems the compilation of this module exceeds the kernel limit. Anybody having any idea?

Thanks

**Comment by wolfzhu** [comment]
Curiously it runs well with tf backend and dumps with plaidml. It looks a pretty small module. Anyone can help me out? Thanks!
